cmd/stellar-rpc/internal/integrationtest/transaction_test.go

```diff
@@ -231,3 +235,107 @@ func TestContractCreationWithConstructor(t *testing.T) {
 	require.NoError(t, err)
 	infrastructure.SendSuccessfulTransaction(t, client, test.MasterKey(), tx)
 }
+
+func TestContractEvents(t *testing.T) {
+	if infrastructure.GetCoreMaxSupportedProtocol() < 23 {
+		t.Skip("Only test this for protocol >= 23")
+	}
+
+	test := infrastructure.NewTest(t, nil)
+	client := test.GetRPCLient()
+
+	// Upload the contract wasm
+	test.UploadEventsContract()
+
+	// Create/deploy an instance of the contract
+	creationOp := infrastructure.CreateCreateEventsContractOperation(test.MasterAccount().GetAccountID())
+	params := infrastructure.PreflightTransactionParams(t, client,
+		infrastructure.CreateTransactionParams(
+			test.MasterAccount(),
+			creationOp,
+		),
+	)
+
+	tx, err := txnbuild.NewTransaction(params)
+	require.NoError(t, err)
+	infrastructure.SendSuccessfulTransaction(t, client, test.MasterKey(), tx)
+
+	// Compute the contract ID based on the deployment operation.
+	preimage := creationOp.HostFunction.MustCreateContractV2().ContractIdPreimage
+	contractID := infrastructure.GetContractID(
+		t,
+		test.MasterAccount().GetAccountID(),
+		preimage.MustFromAddress().Salt,
+		infrastructure.StandaloneNetworkPassphrase,
+	)
+	require.NoError(t, err)
+	params = infrastructure.PreflightTransactionParams(t, client,
+		infrastructure.CreateTransactionParams(
+			test.MasterAccount(),
+			infrastructure.CreateIncrementOperation(
+				xdr.ContractId(contractID),
+				test.MasterAccount().GetAccountID(),
+			),
+		),
+	)
+
+	tx, err = txnbuild.NewTransaction(params)
+	require.NoError(t, err)
+	txsub := infrastructure.SendSuccessfulTransaction(t, client, test.MasterKey(), tx)
+
+	//
+	// Validate the events generated by the invocation
+	//
+
+	events, err := client.GetEvents(t.Context(), protocol.GetEventsRequest{
+		StartLedger: txsub.LatestLedger,
```

Comment:
should this be `txsub.Ledger` to be specific to ledger which had the txsub? it seems like `txsub.LatestLedger` could be susceptible to race, with newer ledger ingestion possibly sneaking in.